logical/framework/field_data_test.go

```diff
@@ -146,6 +146,105 @@ func TestFieldDataGet(t *testing.T) {
 			"foo",
 			0,
 		},
+
+		"slice type, empty slice": {
+			map[string]*FieldSchema{
+				"foo": &FieldSchema{Type: TypeSlice},
+			},
+			map[string]interface{}{
+				"foo": []interface{}{},
+			},
+			"foo",
+			[]interface{}{},
+		},
+
+		"slice type, filled, mixed slice": {
+			map[string]*FieldSchema{
+				"foo": &FieldSchema{Type: TypeSlice},
+			},
+			map[string]interface{}{
+				"foo": []interface{}{123, "abc"},
+			},
+			"foo",
+			[]interface{}{123, "abc"},
+		},
+
+		"string slice type, filled slice": {
```

Comment:
Can we have a test to error out when schema is expecting TypeStringSlice and the raw value contains mixed types just like this test here, but with the differing type of a slice or another JSON blob.